gluten-flink/planner/src/main/java/org/apache/gluten/rexnode/RexNodeConverter.java

```diff
@@ -0,0 +1,114 @@
+import org.apache.calcite.rex.RexLiteral;
+import org.apache.calcite.rex.RexNode;
+
+import java.util.List;
+import java.util.stream.Collectors;
+
+/** Convertor to convert RexNode to velox TypedExpr */
+public class RexNodeConverter {
+
+    public static TypedExpr toTypedExpr(RexNode rexNode) {
+        if (rexNode instanceof RexLiteral) {
+            RexLiteral literal = (RexLiteral) rexNode;
+            return new ConstantTypedExpr(
+                    toType(literal.getType()),
+                    toVariant(literal),
+                    null);
+        } else if (rexNode instanceof RexCall) {
+            RexCall rexCall = (RexCall) rexNode;
+            List<TypedExpr> params = toTypedExpr(rexCall.getOperands());
+            Type nodeType = toType(rexCall.getType());
+            return new CallTypedExpr(
+                    nodeType,
+                    params,
+                    FunctionMappings.toVeloxFunction(rexCall.getOperator().getName()));
+        } else if (rexNode instanceof RexInputRef) {
+            RexInputRef inputRef = (RexInputRef) rexNode;
+            return FieldAccessTypedExpr.create(
+                    toType(inputRef.getType()),
+                    String.valueOf(inputRef.getIndex()));
+        } else {
+            throw new RuntimeException("Unrecognized RexNode: " + rexNode);
+        }
+    }
+
+    public static List<TypedExpr> toTypedExpr(List<RexNode> rexNodes) {
+        return rexNodes.stream()
+                .map(rexNode -> toTypedExpr(rexNode))
+                .collect(Collectors.toList());
+    }
+
+    public static Type toType(RelDataType relDataType) {
+        switch (relDataType.getSqlTypeName()) {
+            case BOOLEAN:
+                return new BooleanType();
+            case INTEGER:
+                return new IntegerType();
+            default:
+                throw new RuntimeException("Unsupported type: " + relDataType.getSqlTypeName());
+        }
+    }
+
+    public static Variant toVariant(RexLiteral literal) {
+        switch (literal.getType().getSqlTypeName()) {
+            case BOOLEAN:
+                return new BooleanValue((boolean) literal.getValue());
+            case TINYINT:
+                return new TinyIntValue(Integer.valueOf(literal.getValue().toString()));
+            case SMALLINT:
+                return new SmallIntValue(Integer.valueOf(literal.getValue().toString()));
+            case INTEGER:
+                return new IntegerValue(Integer.valueOf(literal.getValue().toString()));
+            case BIGINT:
+                return new BigIntValue(Long.valueOf(literal.getValue().toString()));
+            case DOUBLE:
+                return new DoubleValue(Double.valueOf(literal.getValue().toString()));
+            case VARCHAR:
+                return new VarCharValue(literal.getValue().toString());
+            case BINARY:
+                return new VarBinaryValue(literal.getValue().toString());
```

Comment:
Should we use `VarBinaryValue.create()` to pass a byte array in?

Reply:
att